lnwire/error.go

```diff
@@ -47,6 +47,25 @@ func (e FundingError) Error() string {
 	return e.String()
 }
 
+// ErrorLike is an interface implemented by Error and Warning messages as
+// described in BOLT1.
+type ErrorLike interface {
```

Comment:
I don't really see the benefit of this - I dislike changing the peer/brontide code more than it has to be changed. I think it is better if `Warning` does not embed `Error` and either:
- `handleError` is duplicated with a `handleWarning` function
- `handleError` handles both a `Warning` or an `Error` without an interface, and is renamed to `handleErrorOrWarning` or something similar

Reply:
I chose option A :+1: